cpp/jvm-enclave-common/stubs/sys_poll.cpp

```diff
@@ -3,9 +3,12 @@
 //
 #include "vm_enclave_layer.h"
 
-//////////////////////////////////////////////////////////////////////////////
-// Stub functions to satisfy the linker
-STUB(poll);
-
 extern "C" {
+
+    typedef unsigned long int nfds_t;
+
+    int poll(struct pollfd *fds, nfds_t nfds, int timeout) {
+        errno = ENOMEM;
```

Comment:
Did some reading and it looks like we can just return ENOSYS from anything, even if it's not documented for that function.
These all defined so that they have different values, see here:
- https://elixir.bootlin.com/linux/latest/source/include/uapi/asm-generic/errno-base.h#L16
- https://elixir.bootlin.com/linux/latest/source/arch/alpha/include/uapi/asm/errno.h#L51

Reply:
I am still not 100% sure. ENOSYS exists and I think we can probably use. The JVM code does not seem care much about the `errno` value apart from some cases, just to throw the correct exception. There always seem to be a way to handle the returning result (i.e. `-1`), whatever the related `errno` is set to.

Still it is weird that there are some calls for which `ENOSYS` is a valid `errno` value and many others that do not mention it.
I have found this pdf about POSIX standard https://www.open-std.org/jtc1/sc22/open/n4217.pdf  (a bit old), and this suggest that `ENOSYS` has even been removed from some calls ass a valid `errno` value.
Even the Linux kernel does not seem to return `ENOSYS` quite often, see here: `https://github.com/torvalds/linux/blob/master/kernel/time/posix-stubs.c`, where it is used only in one case (not really even sure that that one is a Posix call).

But I have also found a similar case here https://support.kaspersky.com/help/kce/0.0/en-US/posix_uns_ifaces.htm where Kaspersky is doing exactly as you suggest, i.e. return -1 and `ENOSYS` as `errno`.

At the end of the day, what it matters is how the JVM reacts to that and I see that it is always reacting with an error as we want. I I feel that it is fine to do so.